spec/models/appointment_spec.rb

```diff
@@ -0,0 +1,72 @@
+require "rails_helper"
+
+RSpec.describe Appointment, type: :model do
+  let!(:doctor) {FactoryBot.create :doctor}
```

Comment:
let khác gì let! e ?

Reply:
- Let: let chỉ có giá trị trong từng khối it. nếu gọi đến một biến let nhiều lần trong cùng một example thì nó sẽ chỉ khai báo khối let một lần duy nhất ở lần gọi đầu tiên. Đây là khái niệm lazy evaluation. Ví dụ
let(:current_time) { Time.now }
  it "gets current time in it" do
    puts current_time # => 2019-12-26 09:35:29 +0300
    sleep(3)
    puts current_time # => 2019-12-26 09:35:29 +0300
  end
  it "gets the time again" do
    puts current_time # => 2019-12-26 09:35:32 +0300
  end
- Let!: được gọi trong một khối before ngầm, do đó kết quả đã được tính toán và cache lại trước khi khổi it được thực thi. Nên với ví dụ trên khi dùng let! thì với tất cả khối it nào khi 
 puts current_time #=> 2019-12-26 09:35:29 +0300 - giá trị không thay đổi